magento/magento-semver
# MC-38148: Validation of Approved BIC

# Description (*)

This PR contains a collection of changes and fixes as part of the delivery for: https://jira.corp.magento.com/browse/MC-38148

- Adds a new section to HTML SVC report showing the level of changes made to packages in JSON and table format. A Copy to clipboard button is added for the json content.
- Resolves PackageName for the changed file by recursing up the directory to find the relevant composer.json
- Fixes src/Console/Command/CompareSourceCommand.php to return a code instead of exiting
- All Operations (stored versioning changes) will now display the full filepath in the location section of the HTML tables
- Utilizes the mapping array property in the Registry class to store information about the filepath.
- Adds automated testing for testing these changes to the HTML report
  - Copies examples for testing multiple changes together into the tests/Unit/Console/Command/CompareSourceCommandTest/_files/all/ directory



## Fixed Issues (if relevant)


Reviews have been performed within these tickets:
1. https://jira.corp.magento.com/browse/MC-38348
2. https://jira.corp.magento.com/browse/MC-38349


## Manual testing scenarios (*)



1. Run 2.3-develop vs 2.4-develop using both mainline and this branch. Compared changes and make sure they are the same.
2. Ensure package Level Changes displays all package names and level of changes of BIC which causes SVC build fail (core build and MFTF)
3. User should be able to copy content of Package Level Changes and paste to PR description
4. Target/Location column should show file path of the BIC
5. Ensure all unit tests (included new tests) are green 

## Questions or comments



## Contribution checklist (*)
